src/natcap/invest/ui/cbc.py

```diff
@@ -15,28 +14,39 @@ def __init__(self):
             validator=preprocessor.validate,
             localdoc='coastal_blue_carbon.html')
 
-        self.lulc_lookup_uri = inputs.File(
-            args_key='lulc_lookup_uri',
-            helptext=(
-                "A CSV table used to map lulc classes to their values "
-                "in a raster, as well as to indicate whether or not "
-                "the lulc class is a coastal blue carbon habitat."),
-            label='LULC Lookup Table (CSV)',
-            validator=self.validator)
-        self.add_input(self.lulc_lookup_uri)
-        self.lulc_snapshot_list = inputs.Multi(
-            args_key='lulc_snapshot_list',
-            callable_=functools.partial(inputs.File, label="Input"),
-            label='Land Use/Land Cover Rasters (GDAL-supported)',
-            link_text='Add Another')
-        self.add_input(self.lulc_snapshot_list)
+        def _ui_keys(args_key):
```

Comment:
Since this function is repeated in both `CoastalBlueCarbon` and `CoastalBlueCarbonPreprocessor`, could it be defined just one time outside of the classes?

Reply:
424a623a reworks this into a shared function, using `functools.partial` for the instance's reference to the appropriate validator function.

Reply:
Neat!